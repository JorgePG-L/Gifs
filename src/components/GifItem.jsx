import React from 'react'

//import './styles.css'
const GifItem = ({title,urls}) => {
    console.log(title)
  return (
    <div className='card'  >
      <img src= {urls} alt = {title} />
      <p>{title} </p>
    </div>
  )
}



export default GifItem
