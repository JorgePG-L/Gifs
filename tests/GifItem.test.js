

import { render, screen } from "@testing-library/react";
import GifItem from "../src/components/GifItem";

test('should validate snapshot', () => {
    const title = "Saitama";
     const url  = "hhtps://one-punch.com/saitama.jpg";

    const {container} = render(<GifItem title = {title} url = {url} />);
  
    // Utiliza expect junto con toMatchSnapshot
    expect(container).toMatchSnapshot();
})

test('should debe mostrar la imagen con la url y el alt indicado', () => {
    const title = "Saitama";
    const url  = "hhtps://one-punch.com/saitama.jpg";
  render ( <GifItem title = {title} urls = {url} /> );

  const {src,alt} = screen.getByRole('img');

  expect (src).toBe(url);
  expect (alt).toBe(alt);
})

test('should render the title in the component', () => {
    const title = "Saitama";
    const url  = "hhtps://one-punch.com/saitama.jpg";
  render ( <GifItem title = {title} urls = {url} /> );
  expect ( screen.getByText(title)).toBeTruthy();
})
