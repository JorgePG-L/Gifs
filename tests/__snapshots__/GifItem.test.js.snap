// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should validate snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
    />
    <p>
      Saitama
       
    </p>
  </div>
</div>
`;
